frb_codegen/src/parser/ty.rs

```diff
@@ -159,227 +104,368 @@ impl<'a> TypeParser<'a> {
-                    let inner = self.convert_to_ir_type(*generic);
-                    if let Some(IrType::PrimitiveList(IrTypePrimitiveList { primitive })) = inner {
-                        Some(IrType::Delegate(
-                            IrTypeDelegate::ZeroCopyBufferVecPrimitive(primitive),
-                        ))
-                    } else {
-                        None
+
+                    #[cfg(all(feature = "chrono", feature = "qualified_names"))]
+                    [("chrono", None), ("NaiveDateTime", None)] => {
+                        Ok(Delegate(IrTypeDelegate::Time(IrTypeTime::Naive)))
                     }
-                }
-                "RustOpaque" => match *generic {
-                    SupportedInnerType::Array(p, len) => self.convert_array_to_ir_type(*p, len),
-                    SupportedInnerType::Verbatim(ver) => {
-                        Some(IrType::RustOpaque(IrTypeRustOpaque::from(ver.as_ref())))
+
+                    #[cfg(all(feature = "chrono", not(feature = "qualified_names")))]
+                    [("NaiveDateTime", None)] => {
+                        Ok(Delegate(IrTypeDelegate::Time(IrTypeTime::Naive)))
                     }
-                    SupportedInnerType::Path(path) => {
-                        Some(IrType::RustOpaque(IrTypeRustOpaque::from(path.to_string())))
+
+                    #[cfg(feature = "qualified_names")]
+                    [("flutter_rust_bridge", None), ("DartAbi", None)] => {
+                        Ok(Dynamic(IrTypeDynamic))
                     }
-                    SupportedInnerType::Unit => {
-                        Some(IrType::RustOpaque(IrTypeRustOpaque::new_unit()))
+
+                    [("DartAbi", None)] => Ok(Dynamic(IrTypeDynamic)),
+
+                    #[cfg(all(feature = "uuid", feature = "qualified_names"))]
+                    [("uuid", None), ("Uuid", None)] => Ok(Delegate(IrTypeDelegate::Uuid)),
+
+                    #[cfg(all(feature = "uuid", not(feature = "qualified_names")))]
+                    [("Uuid", None)] => Ok(Delegate(IrTypeDelegate::Uuid)),
+
+                    #[cfg(feature = "qualified_names")]
+                    [("flutter_rust_bridge", None), ("DartOpaque", None)] => {
+                        Ok(DartOpaque(IrTypeDartOpaque {}))
                     }
-                },
-                "Box" => self.convert_to_ir_type(*generic).map(|inner| {
-                    Boxed(IrTypeBoxed {
-                        exist_in_real_api: true,
-                        inner: Box::new(inner),
-                    })
-                }),
-                "Option" => {
-                    // Disallow nested Option
-                    if matches!(generic.as_ref(), SupportedInnerType:: Path(SupportedPathType { ident, .. }) if ident == "Option")
+
+                    [("DartOpaque", None)] => Ok(DartOpaque(IrTypeDartOpaque {})),
+
+                    [("String", None)] => Ok(Delegate(IrTypeDelegate::String)),
```

Comment:
uber-nit: is it possible we avoid duplicating code for qualified_names? For example, is it possible we write down something like (super-pseudo code, hope you get the idea):

```rust
struct MyMatcher { crate_name, class_name }

impl MyMatcher {
  fn match(&self, flat_array) { flat_array == [crate_name, class_name] || flat_array == [class_name] }
}

let uuid_matcher = MyMatcher("uuid", "Uuid")

uuid_matcher.match(flat_array)
```

then no need to duplicate "uuid::Uuid" and "Uuid" as two separate cases

Reply:
Would structural pattern matching still be possible? I think what we'd gain in DRYness we would lose tenfold in readability.

Reply:
Hmm looks reaasonable!

Reply:
For what it's worth, you could mark `not(feature = "qualified_names")` deprecated and clean them up for the next major version.